truffle/contracts/Auction.sol

```diff
@@ -36,27 +36,32 @@ contract Auction is ReentrancyGuard {
     address public highestBidder;
     mapping(address => uint) public bids;
 
-    constructor () {
-        seller = payable(msg.sender);
+    constructor(
+        address sender,
+        IERC721 _nft,
+        uint _nftId,
+        uint _startingBid,
+        uint _increment
+    ) {
+        require(_startingBid > 0, "Starting bid must be greater than 0!");
+        require(_increment > 0, "Increment must be greater than 0!");
+        seller = payable(sender);
+        highestBid = _startingBid;
+        increment = _increment;
+        nft = _nft;
+        nftId = _nftId;
     }
 
-    function start(IERC721 _nft, uint _nftId, uint startingBid, uint _increment, uint _duration) external {
+    function start(uint _duration, uint _startTime) external {
         require(!started, "Auction already started!");
-        require(msg.sender == seller, "You did not start the auction!");
+        require(msg.sender == seller, "You are not the owner of this auction!");
         require(_duration > 0, "Duration must be greater than 0!");
-        require(increment > 0, "Increment must be greater than 0!");
-        require(startingBid > 0, "Starting bid must be greater than 0!");
-        highestBid = startingBid;
 
-        nft = _nft;
-        nftId = _nftId;
         nft.transferFrom(msg.sender, address(this), nftId);
-
         started = true;
-        startAt = block.timestamp;
-        increment = _increment;
+        startAt = _startTime;
```

Comment:
Actually hmm, i guess can use either block timestamp or user input uh. Also, reason why i left the startTime and duration in the start is becoz the I tot it makes more sense. Like when they create an auction, they just specify info regarding the auction like the bid, bid amt etc. Between start and auction creation, user has to go to the nft contract to approve, after which they can start bid anytime. If we put start and duration in constructor, then user have to approve by a certain time and then start the bid.

Reply:
I see, was thinking combining it could save gas. I think the user input still has to be validated now to ensure it is not in the past and not after the end time

Reply:
ahh i see, didnt rly think abt the gas aspect lool. But okie i've added the check in alr